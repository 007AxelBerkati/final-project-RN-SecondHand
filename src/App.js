import CodePush from 'react-native-code-push';
import FlashMessage from 'react-native-flash-message';
import { Provider, useSelector, useDispatch } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import React, { useEffect } from 'react';
import Router from '@router';

import { LogBox, StatusBar } from 'react-native';
import {
  getNotifikasi, Persistore, Store,
} from './redux';
import { Loading } from './components';

const CodePushOptions = {
  checkFrequency: CodePush.CheckFrequency.ON_APP_START,
  mandatoryInstallMode: CodePush.InstallMode.IMMEDIATE,
  updateDialog: {
    appendReleaseDescription: true,
  },
};

function MainApp() {
  const stateGlobal = useSelector((state) => state.dataGlobal);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getNotifikasi());
  }, []);

  return (
    <>
      <StatusBar barStyle="dark-content" />
      <Router />
      <FlashMessage position="top" />
      {stateGlobal.isLoading && <Loading />}
    </>
  );
}

function App() {
  useEffect(() => {
    LogBox.ignoreLogs(['VirtualizedLists should never be nested']);
  }, []);
  return (
    <Provider store={Store}>
      <PersistGate loading={null} persistor={Persistore}>
        <MainApp />
      </PersistGate>
    </Provider>
  );
}

export default CodePush(CodePushOptions)(App);
