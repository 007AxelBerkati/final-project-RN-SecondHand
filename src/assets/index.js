export * from './icons';
export * from './images';
export * from './dummyData';
