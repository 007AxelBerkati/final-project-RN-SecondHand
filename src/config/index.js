// eslint-disable-next-line import/no-cycle
import instance from './instance';

export { instance };
