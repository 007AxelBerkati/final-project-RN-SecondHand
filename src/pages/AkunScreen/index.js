import {
  ScrollView, StyleSheet, Text, View,
} from 'react-native';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Fade, Placeholder, PlaceholderMedia } from 'rn-placeholder';
import {
  CardList, Headers, NotLogin, Profile,
} from '../../components';
import {
  borderRadius,
  colors, fonts, fontSize, windowHeight, windowWidth,
} from '../../utils';
import { version } from '../../../package.json';
import { getAkun, logout } from '../../redux';

function AkunScreen({ navigation }) {
  const dispatch = useDispatch();
  const dataProfile = useSelector((state) => state.dataProfile);
  const dataLogin = useSelector((state) => state.dataLogin);

  useEffect(() => {
    dispatch(getAkun());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const onLogout = () => {
    dispatch(logout());
    navigation.replace('MainApp');
  };

  return (
    <View style={styles.pages}>
      <Headers title="Akun Saya" />
      {
        !dataLogin.isLoggedIn ? (
          <NotLogin onPress={() => navigation.navigate('LoginScreen')} />
        ) : (
          <ScrollView>
            {
              dataProfile.isLoading ? (
                <Placeholder
                  Animation={Fade}
                  style={styles.photoSection}
                >
                  <PlaceholderMedia style={styles.placeholder} />
                </Placeholder>
              ) : (
                <Profile source={{ uri: dataProfile.profile?.image_url }} />
              )
            }
            <CardList type="account" name="edit" title="Ubah Akun" onPress={() => navigation.navigate('ProfileScreen')} />
            <CardList type="account" name="setting" title="Pengaturan Akun" onPress={() => navigation.navigate('PengaturanScreen')} />
            <CardList type="account" name="logout" title="Keluar" onPress={onLogout} />
            <Text style={styles.version}>
              Version
              {' '}
              {version}
            </Text>
          </ScrollView>
        )

      }
    </View>
  );
}

export default AkunScreen;

const styles = StyleSheet.create({
  pages: {
    flex: 1,
    margin: 16,
  },

  version: {
    fontFamily: fonts.Poppins.Regular,
    fontSize: fontSize.small,
    color: colors.text.subtitle,
    marginTop: 16,
    alignSelf: 'center',
  },

  placeholder: {
    borderRadius: borderRadius.xlarge,
    width: windowWidth * 0.3,
    height: windowHeight * 0.15,
    alignSelf: 'center',
  },

  photoSection: {
    alignItems: 'center',
    marginVertical: 24,
  },
});
