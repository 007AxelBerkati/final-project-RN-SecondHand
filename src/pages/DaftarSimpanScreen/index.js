import { View } from 'react-native';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import CardProduct from '../../components/molekuls/CardProduct';
import { Headers } from '../../components';
import { getWishlistBuyer } from '../../redux';

function DaftarSimpanScreen({ navigation }) {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.data);
  useEffect(() => {
    dispatch(getWishlistBuyer);
  });

  return (
    <View style={{ flex: 1, margin: 16 }}>
      <Headers type="back-title" onPress={() => navigation.goBack()} title="Daftar Simpan" />
      {data?.Product?.map((item) => (
        <CardProduct source={item.image_url} harga={item.base_price} name={item.name} icon="bookmark" />
      ))}

    </View>
  );
}

export default DaftarSimpanScreen;
