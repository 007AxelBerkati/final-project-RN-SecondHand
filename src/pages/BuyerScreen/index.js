import {
  SafeAreaView, ScrollView, StyleSheet, View,
  StatusBar, Text,
} from 'react-native';
import React, { useCallback, useMemo, useRef } from 'react';
import { ImageSlider } from 'react-native-image-slider-banner';
import BottomSheetModal from '@gorhom/bottom-sheet';
import {
  ButtonComponent, CardList, CardProduct, Desc, Gap,
} from '../../components';
import { keranjang } from '../../assets';
import {
  colors, windowHeight, windowWidth,
} from '../../utils';

function BuyerScreen() {
  const bottomSheetModalRef = useRef(null);

  // variables
  const snapPoints = useMemo(() => ['25%', '50%'], []);

  // callbacks
  const handlePresentModalPress = useCallback(() => {
    bottomSheetModalRef.current?.present();
  }, []);
  const handleSheetChanges = useCallback((index) => {
    console.log('handleSheetChanges', index);
  }, []);
  return (
    <SafeAreaView style={styles.pages}>
      <ScrollView showsVerticalScrollIndicator>
        <StatusBar backgroundColor="transparent" translucent />
        <View style={styles.imageContainer}>
          <ImageSlider
            data={[
              { img: 'https://img.freepik.com/free-vector/online-shopping-concept-illustration_114360-1084.jpg?w=1060&t=st=1655909482~exp=1655910082~hmac=17398ae027564d6d34625f5b83f8ab39616da2ea74d55677a11eb56ce8eb70bd' },
              { img: 'https://media.istockphoto.com/photos/male-outfit-set-on-white-background-closeup-top-view-picture-id1176614800?k=20&m=1176614800&s=612x612&w=0&h=uM0CkYhsM12_OYWhnO641tXqD3ZCOx61dyGZClkrv6k=' },
              { img: 'https://images.unsplash.com/photo-1479064555552-3ef4979f8908?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=870&q=80' },
            ]}
            autoPlay
            timer={5000}
            closeIconColor={colors.background.primary}
            caroselImageStyle={{ height: windowHeight * 0.4 }}
            indicatorContainerStyle={{ bottom: windowHeight * 0.05 }}
          />
          <View style={styles.btnBackContainer}>
            <ButtonComponent
              type="icon-button"
              label="BackButton"
            />
          </View>
          <View style={styles.productWrapper}>
            <CardProduct name="Jam tangan" jenis="Barang Antik" harga="10000" />
          </View>
        </View>
        <View style={styles.card}>
          <CardList
            source={keranjang}
            name="Nama Penjual"
            kota="palangka"
            type="role"
          />
        </View>
        <View style={styles.title}>
          <Desc label="Deskripsi" desc="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum." />
        </View>
        <Gap height={60} />
      </ScrollView>
      <View style={styles.btnNego}>
        <ButtonComponent title="saya tertarik dan ingin nego" onPress={handlePresentModalPress} />
      </View>
      <BottomSheetModal
        ref={bottomSheetModalRef}
        index={1}
        snapPoints={snapPoints}
        onChange={handleSheetChanges}
      >
        <View style={styles.contentContainer}>
          <Text>Awesome ðŸŽ‰</Text>
        </View>
      </BottomSheetModal>
    </SafeAreaView>
  );
}

export default BuyerScreen;

const styles = StyleSheet.create({
  imageContainer: {
    height: windowHeight * 0.5,
    width: windowWidth,
  },

  btnBackContainer: {
    position: 'absolute',
    top: 44,
    left: 16,
  },

  title: {
    marginTop: -12,
  },

  card: {
    marginHorizontal: 16,
    marginBottom: 30,
    marginTop: -24,
  },

  pages: {
    flex: 1,
    backgroundColor: colors.background.primary,
  },

  btnNego: {
    padding: 16,
    position: 'absolute',
    width: '100%',
    bottom: 1,
  },

  productWrapper: {
    marginHorizontal: 16,
    marginTop: -40,
  },

});
