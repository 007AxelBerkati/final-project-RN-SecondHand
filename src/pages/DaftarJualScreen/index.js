import {
  StyleSheet, View,
  RefreshControl,
} from 'react-native';
import React, { useCallback, useEffect, useState } from 'react';
import { ScrollView } from 'react-native-gesture-handler';
import { useSelector, useDispatch } from 'react-redux';
import {
  CardCategory, CardList, Headers, NotLogin,
} from '../../components';
import { listDaftarJual } from '../../assets';
import {
  getAkun, getOrderSeller, getProductSeller,
} from '../../redux';
import {
  colors, fonts, fontSize, windowHeight,
} from '../../utils';
import Produk from './Produk';
import Favorite from './Favorite';
import Terjual from './Terjual';

function DaftarJualScreen({ navigation }) {
  const [active, setActive] = useState(1);
  const dispatch = useDispatch();
  const [refreshing, setRefreshing] = useState(false);

  const dataDaftarJual = useSelector((state) => state.dataDaftarJual.daftarJual);
  const dataLogin = useSelector((state) => state.dataLogin);
  const dataProfile = useSelector((state) => state.dataProfile.profile);
  const dataOrder = useSelector((state) => state.dataDaftarJual.productDiminati);

  const getDaftarJual = (id) => {
    setActive(id);
    switch (id) {
      case 1:
        dispatch(getProductSeller());
        break;
      case 2:
        dispatch(getOrderSeller());
        break;
      case 3:
        dispatch(getOrderSeller());
        break;
      default:
        break;
    }
  };

  useEffect(() => {
    dispatch(getAkun());
    dispatch(getProductSeller());
  }, []);

  const onRefresh = useCallback(async (id) => {
    setRefreshing(true);
    getDaftarJual(id);
    setRefreshing(false);
  }, []);

  const dataForRender = () => {
    switch (active) {
      case 1:
        return <Produk dataDaftarJual={dataDaftarJual} navigation={navigation} />;
      case 2:
        return <Favorite productDiminati={dataOrder} navigation={navigation} />;
      case 3:
        return <Terjual dataTerjual={dataOrder} navigation={navigation} />;
      default:
        return null;
    }
  };

  return (
    <View style={styles.pages}>
      <Headers title="Daftar Jual Saya" />
      {
        !dataLogin.isLoggedIn ? (
          <NotLogin onPress={() => navigation.navigate('LoginScreen')} />
        ) : (
          <ScrollView
            showsVerticalScrollIndicator={false}
            refreshControl={(
              <RefreshControl
                refreshing={refreshing}
                onRefresh={() => onRefresh(active)}
              />
            )}
          >
            <View style={{ marginHorizontal: 3 }}>
              <CardList type="role" name={dataProfile.full_name} source={{ uri: dataProfile.image_url }} kota={dataProfile.city} onPress={() => navigation.navigate('ProfileScreen')} />
              <ScrollView
                horizontal
                showsHorizontalScrollIndicator={false}
                style={{ marginVertical: 24 }}
              >
                {
                  listDaftarJual.data.map((item) => (
                    <View key={item.id}>
                      <CardCategory
                        active={active === item.id}
                        name={item.name}
                        kategori={item.category}
                        onPress={() => getDaftarJual(item.id)}
                      />
                    </View>
                  ))
                }
              </ScrollView>
              {dataForRender()}
            </View>
          </ScrollView>
        )
      }

    </View>
  );
}

export default DaftarJualScreen;

const styles = StyleSheet.create({
  pages: {
    flex: 1,
    margin: 16,
  },

  notLogin: {
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: 24,
  },

  image: {
    width: '100%',
    height: windowHeight / 3,
    resizeMode: 'contain',
    marginTop: windowHeight * 0.15,
  },

  notLoginText: {
    fontFamily: fonts.Poppins.Medium,
    fontSize: fontSize.medium,
    color: colors.text.primary,
  },

});
