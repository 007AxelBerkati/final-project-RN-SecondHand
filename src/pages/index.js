import LoginScreen from './LoginScreen';
import RegisterScreen from './RegisterScreen';
import HomeScreen from './HomeScreen';
import DaftarJualScreen from './DaftarJualScreen';
import AkunScreen from './AkunScreen';
import JualScreen from './JualScreen';
import NotifikasiScreen from './NotifikasiScreen';
import SplashScreen from './SplashScreen';
import InfoPenawaranScreen from './InfoPenawaranScreen';

export {
  LoginScreen,
  RegisterScreen, HomeScreen, DaftarJualScreen, AkunScreen, JualScreen, NotifikasiScreen,
  SplashScreen, InfoPenawaranScreen,
};
