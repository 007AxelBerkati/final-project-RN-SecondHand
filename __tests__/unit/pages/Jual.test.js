import { render } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux';
import { JualScreen } from '../../../src/pages';
import { Store } from '../../../src/redux';

describe('JualScreen', () => {
  it('should render JualScreen', () => {
    const tree = render(
      <Provider store={Store}>
        <JualScreen />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
