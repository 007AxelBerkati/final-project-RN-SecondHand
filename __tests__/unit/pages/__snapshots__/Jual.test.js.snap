// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JualScreen should render JualScreen 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "marginTop": 40.019999999999996,
        "width": null,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "left": 0,
          "opacity": 1,
          "position": "absolute",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#112340",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ÔÑê
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#112340",
          "fontFamily": "Poppins-Bold",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Lengkapi Detail Produk
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 24,
        }
      }
    >
      <test-file-stub
        style={
          Object {
            "height": 667,
            "resizeMode": "contain",
            "width": 600,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#7D8797",
            "fontFamily": "Poppins-Medium",
            "fontSize": 22,
            "textAlign": "center",
          }
        }
      >
        Anda belum login, Silahkan login terlebih dahulu
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FF8303",
          "borderColor": "#FF8303",
          "borderRadius": 18.857142857142858,
          "borderWidth": 1,
          "bottom": 15,
          "left": 0,
          "opacity": 1,
          "paddingVertical": 10,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Poppins-Regular",
            "fontSize": 25.142857142857142,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
